Client/Frontend/Browser/PasswordHelper.swift

```diff
@@ -88,14 +88,15 @@ class PasswordHelper: BrowserHelper {
 
         profile.passwords.get(QueryOptions(filter: password.hostname), complete: { data in
             for i in 0..<data.count {
-                let savedPassword = data[i] as! Password
-                if savedPassword.username == password.username {
-                    if savedPassword.password == password.password {
+                if let savedPassword = data[i] {
```

Comment:
We should probably make this a "for password in data", but I remember hitting some swift errors originally when i implemented that maybe it doesn't work. Still data[i] should not be nil here. If it is, something is busted.